include/deal.II/numerics/vector_tools.h

```diff
@@ -857,6 +858,59 @@ namespace VectorTools
                 const bool                          project_to_boundary_first = false);
 
   /**
+   * The same as above for projection of scalar-valued quadrature data in case
+   * of parallel Triangulations.
+   * The underlying MatrixFree implementation requires to specify the degree of
+   * the used FiniteElement as additional template parameters.
+   * The user provided function should return a value at the quadrature point
+   * based on the cell iterator and quadrature number and of course should be
+   * consistent with the provided @p quadrature object, which will be used
+   * to assemble the right-hand-side.
+   *
+   * This function can be used with lambdas:
+   * @code
+   * VectorTools::project_parallel<dim,LA::MPI::Vector,dim,fe_degree>
+   * (mapping,
+   *  dof_handler,
+   *  constraints,
+   *  quadrature_formula,
+   *  [=] (const typename DoFHandler<dim>::active_cell_iterator & cell, const unsigned int q) -> double { return qp_manager.get_data(cell)[q]->density; },
+   *  field);
+   * @endcode
+   */
+  template <int dim, typename VectorType, int spacedim, int fe_degree>
+  void project_parallel (const Mapping<dim, spacedim>   &mapping,
+                         const DoFHandler<dim,spacedim> &dof,
+                         const ConstraintMatrix         &constraints,
+                         const Quadrature<dim>          &quadrature,
+                         const std::function< typename VectorType::value_type (const typename DoFHandler<dim, spacedim>::active_cell_iterator &, const unsigned int)> func,
+                         VectorType                     &vec_result);
+
+  /**
+   * The same as above for projection of scalar-valued MatrixFree quadrature
+   * data with parallel Triangulation.
+   * The underlying MatrixFree implementation requires to specify the degree of
+   * the used FiniteElement and the number of quadrature points in 1D as
+   * additional template parameters.
+   * The user provided function @p func should return a Vectorized value at the
+   * quadrature point based on the cell number and quadrature number.
+   *
+   * This function can be used with lambdas:
+   * @code
+   * VectorTools::project_parallel<dim,LinearAlgebra::distributed::Vector<double>,fe_degree, n_q_points_1d>
+   * (matrix_free_data,
+   *  constraints,
+   *  [=] (const unsigned int cell, const unsigned int q) -> VectorizedArray<double> { return qp_data(cell,q); },
```

Comment:
same here -- it would be good to have a brief explanation of what `qp_data` might refer to.